kobotoolbox/kpi
# feat(projectHistoryLogs): log duplicated submissions TASK-1342

### 🗒️ Checklist

1. [x] run linter locally
2. [x] update all related docs (API, README, inline, etc.), if any
3. [x] draft PR with a title `<type>(<scope>)<!>: <title> TASK-1234`
4. [x] tag PR: at least `frontend` or `backend` unless it's global
5. [x] fill in the template below and delete template comments
6. [x] review thyself: read the diff and repro the preview as written
7. [ ] open PR & confirm that CI passes
8. [x] request reviewers, if needed
9. [ ] delete this section before merging

### 📣 Summary
Record when submissions are duplicated. Record it just as a regular submission.

### 💭 Notes
Add submission information to the request, then process as usual in `ProjectHistoryLogs.create_from_request`.


### 👀 Preview steps

1. ℹ️ have at least 2 accounts and a project. Log in to the first account
2. Give the second account the Manage Project permission
3. add a submission to the project and take note of the id
4. from the terminal, run `curl -X POST -H 'Authorization: Token <account1_token>'  kf.kobo.local/api/v2/assets/<asset_uid>/data/<submission_id>/duplicate/`
5. reload the data table to make sure the submission was duplicated
6. go to `api/v2/assets/<asset_uid>/history`
7. 🟢 there should be a new PH log with action=`add-submission` and the usual metadata, plus
```
"submission": {
    "submitted_by": <account1_username>
}
```
8. from the terminal, run `curl -X POST -H 'Authorization: Token <account2_token>'  kf.kobo.local/api/v2/assets/<asset_uid>/data/<submission_id>/duplicate/` (notice this is using account2's token)
9. reload the data table to make sure the submission was duplicated again, with the new submission's "submitted_by" being account2.
10. go to `api/v2/assets/<asset_uid>/history`
11. 🟢 there should be a new PH log with action=`add-submission` and the usual metadata, plus
```
"submission": {
    "submitted_by": <account2_username>
}
```
